x-pack/plugins/ingest_pipelines/public/application/components/pipeline_editor/components/processor_form/processors/network_direction.tsx

```diff
@@ -0,0 +1,264 @@
+    },
+    config: {
+      type: FIELD_TYPES.COMBO_BOX,
+      deserializer: to.arrayOfStrings,
+      serializer: from.optionalArrayOfStrings,
+      fieldsToValidateOnChange: ['fields.internal_networks', 'fields.internal_networks_field'],
+      validations: [
+        {
+          validator: ({ value, path, formData }) => {
+            if (isEmpty(value) && isEmpty(formData['fields.internal_networks_field'])) {
+              return { path, message: 'A field value is required.' };
+            }
+          },
+        },
+      ],
+      label: i18n.translate(
+        'xpack.ingestPipelines.pipelineEditor.networkDirection.internalNetworksLabel',
+        {
+          defaultMessage: 'Internal networks',
+        }
+      ),
+      helpText: (
+        <FormattedMessage
+          id="xpack.ingestPipelines.pipelineEditor.networkDirection.internalNetworksHelpText"
+          defaultMessage="List of internal networks."
+        />
+      ),
+    },
+  },
+  internal_networks_field: {
+    path: 'fields.internal_networks_field',
+    config: {
+      type: FIELD_TYPES.TEXT,
+      serializer: from.emptyStringToUndefined,
+      fieldsToValidateOnChange: ['fields.internal_networks', 'fields.internal_networks_field'],
+      validations: [
+        {
+          validator: ({ value, path, formData }) => {
+            if (isEmpty(value) && isEmpty(formData['fields.internal_networks'])) {
+              return { path, message: 'A field value is required.' };
+            }
+          },
+        },
+      ],
+      label: i18n.translate(
+        'xpack.ingestPipelines.pipelineEditor.networkDirection.internalNetworksFieldLabel',
+        {
+          defaultMessage: 'Internal networks field',
+        }
+      ),
+      helpText: (
+        <FormattedMessage
+          id="xpack.ingestPipelines.pipelineEditor.networkDirection.internalNetworksFieldHelpText"
+          defaultMessage="Field stating where to read the {field} configuration from."
+          values={{
+            field: <EuiCode>{'internal_networks'}</EuiCode>,
+          }}
+        />
+      ),
+    },
+  },
+};
+
+export const NetworkDirection: FunctionComponent = () => {
+  const [{ fields }] = useFormData({ watch: 'fields.internal_networks_field' });
+  const [isCustom, setIsCustom] = useState<boolean | undefined>();
+
+  // On initial render the fields variable is undefined and eventually get loaded
+  // with data as the form-lib fields get rendered. Since the state of the UI for the
+  // `NetworksFields` depends on that, we need to have an effect that runs only once
+  // when the `fields` have been loaded and when the isCustom hasnt been yet set.
+  const internalNetworksFieldValue = fields?.internal_networks_field;
+  useEffect(() => {
+    if (internalNetworksFieldValue && isCustom === undefined) {
+      setIsCustom(internalNetworksFieldValue?.length > 0);
+    }
+  }, [internalNetworksFieldValue, isCustom]);
+
```

Comment:
Ended up not doing an early return based on `internalNetworksFieldValue` because if we’re creating a new processor the field will be undefined. If we still wanna do an early return based on fields, we need to remove the `watch` prop from the `useFormData` hook since if we're then creating a new processor it will be undefined and nothing will get rendered.